lms/services/grouping.py

```diff
@@ -32,52 +33,49 @@ def generate_authority_provided_id(
             tool_consumer_instance_guid, parent.lms_id, type_.value, lms_id
         )
 
-    def upsert_with_parent(  # pylint: disable=too-many-arguments
-        self,
-        tool_consumer_instance_guid,
-        lms_id,
-        lms_name,
-        parent: Grouping,
-        type_: Grouping.Type,
-        extra=None,
-    ):
+    class _GroupingData(TypedDict):
```

Comment:
This might be a bit meh I still prefer it over just a list of random dicts, it's the same in practice but IMO this more explicit than a long comment.